src/api.rs

```diff
@@ -176,10 +178,25 @@ pub enum Registers {
 pub const PAGE_SHIFT: u32 = 12;
 pub const PAGE_SIZE: u32 = 4096;
 
+#[derive(thiserror::Error, Debug)]
+pub enum MicrovmiError {
+    #[error(transparent)]
```

Comment:
So, in the talk you linked in my PR this behavior is essentially regarded as a bug (if I'm not mistaken) because it messes with proper error reporting.